src/torchcodec/decoders/_core/VideoDecoderOps.cpp

```diff
@@ -63,13 +65,14 @@ TORCH_LIBRARY(torchcodec_ns, m) {
 }
 
 namespace {
+
 at::Tensor wrapDecoderPointerToTensor(
     std::unique_ptr<VideoDecoder> uniqueDecoder) {
   VideoDecoder* decoder = uniqueDecoder.release();
 
   auto deleter = [decoder](void*) { delete decoder; };
   at::Tensor tensor =
-      at::from_blob(decoder, {sizeof(VideoDecoder)}, deleter, {at::kLong});
+      at::from_blob(decoder, {sizeof(VideoDecoder*)}, deleter, {at::kLong});
```

Comment:
Drive-by: we have been allocating enough space for an entire `VideoDecoder` _object_. But we're not storing the entire object, just a pointer.